### PR TITLE
Add hover event support to the engine

### PR DESCRIPTION
**Pull request by gspencergoog**
This adds support for passing ACTION_HOVER_MOVE events to the framework, in preparation for supporting mouse hover events on Android and ChromeOS.

### CODE REVIEW

File: shell/platform/android/io/flutter/view/FlutterView.java [hunk 1]
```diff
@@ -557,7 +559,7 @@ public boolean onTouchEvent(MotionEvent event) {
             }
         }
 
-        assert packet.position() % (kPointerDataFieldCount * kBytePerField) == 0;
+        assert packet.position() % (kPointerDataFieldCount * kPointerBytesPerField) == 0;
```
Comment (tvolkert) [inline shell/platform/android/io/flutter/view/FlutterView.java]: Just filed https://github.com/flutter/flutter/issues/25517

Comment (gspencergoog) [reply shell/platform/android/io/flutter/view/FlutterView.java]: Ahh, OK, I'll fix that too in this PR then.

**Comment by chinmaygarde** [comment]
Is this good to go? Trying to clear the PR queue.

**Comment by gspencergoog** [comment]
Yes, it's good to go. I'll commit.

**Comment by chinmaygarde** [comment]
Thanks!

**Pull request merged**